app/controllers/application_controller.rb

```diff
@@ -1,5 +1,5 @@
 class ApplicationController < ActionController::Base
-    def not_found
-        raise ActionController::RoutingError.new('Not Found')
+    rescue_from ActiveRecord::RecordNotFound do |_exception|
```

Comment:
This should be read before committing to this change.

https://edgeguides.rubyonrails.org/action_controller_overview.html#rescue

Reply:
Hmmm, if I'm understanding correctly, in development it should automatically be throwing a 404? In that case I may have some tinkering to do with my rspec setup.

Reply:
Did you read this part? 
![image](https://user-images.githubusercontent.com/14803/166629486-72e12b6b-ba5a-4734-8600-7349a7898621.png)

Reply:
Also, this hints at the fact that it is handled by default.

![image](https://user-images.githubusercontent.com/14803/166629544-fdfa659d-bdc8-4f70-92b1-8001d7ae2a31.png)

Reply:
Oops, read the wrong portion :laughing: , refactored with the better pattern. Thanks again!

Reply:
Here is the way I see it. Rails takes care of showing a 404 in production by default. This means that you probably don't have to write a test for it. That doesn't mean it shouldn't be considered. In the case that we can trust our framework, we should do that and reduce the number of tests we write as a result.

If you decide that you want to do more than show the 404.html in the public directory, then you can rescue from the error yourself. If you did this, then I would probably write a test for it.

I brought it up earlier in the review as a general "What are we testing?" answer. My opinion is that we can say we need our app to return a 404 when the resource isn't found, if we are confident that Rails does this for us already then we don't need to write tests for it. If we don't know how 404s are handled then the research has to be done to get to the point where we are confident that it's being handled. Maybe other frameworks don't handle it the same way and then tests would be necessary. Don't want to redo work the framework already does, even though it can be hard to keep track of what Rails is responsible for.

Reply:
I like it 🤔 , appreciate you giving me a line of thinking to go down instead of a simple "we don't need this test" 😄 .
Initially I read that 404's are handled in production by default but that made me think, ah so we have to write tests just for development? Or do we have to tinker with the dev environment so all requests follow the same behavior as production.

Thanks for the explanation, I think it's fair to lean on the default behavior now that I understand it better 😆 .

Reply:
Yeah sounds good. We can trust rails to handle RecordNotFound in production, so a test isn't necessary unless we overwrite that default behaviour with our own code.